src/sentry/migrations/0232_backfill_missed_semver_releases.py

```diff
@@ -0,0 +1,154 @@
+# Generated by Django 2.2.24 on 2021-09-23 14:02
+from datetime import datetime
+
+import pytz
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+from sentry_relay.exceptions import RelayError
+from sentry_relay.processing import parse_release
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+
+
+def convert_build_code_to_build_number(build_code):
+    # Taken from `ReleaseModelManager._convert_build_code_to_build_number`
+    build_number = None
+    if build_code is not None:
+        try:
+            build_code_as_int = int(build_code)
+            if validate_bigint(build_code_as_int):
+                build_number = build_code_as_int
+        except ValueError:
+            pass
+    return build_number
+
+
+def validate_bigint(value):
+    return isinstance(value, int) and value >= 0 and value.bit_length() <= 63
+
+
+UPDATE_QUERY = """
+    UPDATE sentry_release
+    SET package = data.package,
+    major = data.major::bigint,
+    minor = data.minor::bigint,
+    patch = data.patch::bigint,
+    revision = data.revision::bigint,
+    prerelease = data.prerelease,
+    build_code = data.build_code,
+    build_number = data.build_number::bigint
+    FROM (VALUES %s) AS data (id, package, major, minor, patch, revision, prerelease, build_code, build_number)
+    WHERE sentry_release.id = data.id"""
+
+SEMVER_REGEX = (
+    r"^[^@\/\s\"''\[\]]+@([0-9]([0-9]{0,10}|[0-9]{12,25}))(\.([0-9][0-9]*))?"
+    r"(\.([0-9][0-9]*))?(\.([0-9][0-9]{*))?(-(0|[1-9][0-9]*|[0-9]*[a-zA-Z-]"
+    r"[0-9a-zA-Z-]*)?(\.(0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*)?"
+    r"(\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+)
+SEMVER_FIELDS = ["package", "major", "minor", "patch", "revision", "prerelease", "build_code"]
+
+
+def backfill_semver(apps, schema_editor):
+    Release = apps.get_model("sentry", "Release")
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        Release.objects.filter(
+            version__contains="@",
+            version__regex=SEMVER_REGEX,
+            major=None,
+            date_added__gte=datetime(2021, 7, 21, tzinfo=pytz.utc),
+        ).values_list(
```

Comment:
I'd recommend we just iterate over the whole table like the previous backfills do. `2021/07/21` is only applicable to getsentry and not to on-prem/st users, and that regex is pretty expensive to run, and will run for each batch that `RangeQuerySetWrapperWithProgressBar` processes.

It'd be fine to just entirely duplicate src/sentry/migrations/0223_semver_backfill_2.py as is and just have ops run that again.

Reply:
@wedamija what if we narrow it down just by keeping `major=None` filter, surely it is not needed to re-process the already parsed ones .. that brings us down to about 6.5 mill releases

Reply:
Scratch that, that's incorrect.. difference is not that massive

Reply:
@ahmedetefy The other thing is that when we use `RangeQuerySetWrapperWithProgressBar` it orders by id so that we can iterate through in batches. So typically the index used will be on pk, and we won't be able to make use of other indexes.

In practice, for this table it's not a big deal. This backfill takes < 1 day to run from what I remember, and it's written fairly efficiently since it only makes an update if the parsed fields change, and batches updates in groups of 100.

Reply:
You are right